restcomm/restcomm.application/src/main/webapp/WEB-INF/scripts/mariadb/sql/permissions.xml

```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE mapper PUBLIC "-//mybatis.org//DTD Mapper 3.0//EN" "http://mybatis.org/dtd/mybatis-3-mapper.dtd">
+<!--
+  @author
+-->
+<mapper namespace="org.mobicents.servlet.sip.restcomm.dao.PermissionsDao">
+  <insert id="addPermission" parameterType="map">
+    INSERT INTO restcomm_permissions (sid, name) VALUES(#{sid}, #{name});
+  </insert>
+
+  <select id="getPermission" parameterType="string" resultType="hashmap">
+    SELECT * FROM restcomm_permissions WHERE sid=#{sid};
+  </select>
+  
+  <select id="getPermissionByName" parameterType="string" resultType="hashmap">
+    SELECT * FROM restcomm_permissions WHERE name=#{name};
+  </select>
+
+  <delete id="deletePermission" parameterType="string">
+    DELETE FROM restcomm_permissions WHERE sid=#{sid};
```

Comment:
what happens with associated accountPermissiosn poiting to this permission?Is DB caring about removing those? Where is the cascading configured?
I expect removing permission with account associations to be prevented until no more associatitions are there?

Reply:
Theres no cascade/constraints enforced. If the permission doesnt exist, the user will get a null in the code (from the join table) and wont be granted permissions
Additionally, theres no checks to ensure the existence of the Permission before adding it in the join table.
Not a big fan of cascading deletes/modifies, but if you think we should have FK constraints we can add them